src/units/attack_type.hpp

```diff
@@ -125,6 +125,13 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 
 	// Configured as a bit field, in case that is useful.
 	enum AFFECTS { AFFECT_SELF=1, AFFECT_OTHER=2, AFFECT_EITHER=3 };
+	/** overwrite_special_checking : return an unit_ability_list list after checking presence or not of overwrite_specials
+	 * @param ability The special ability type who is being checked.
+	 * @param temp_list the list checked and returned.
+	 * @param abil_list list checked for verify presence of overwrite_specials .
+	 * @param filter_self name of [filter_"self/student"] if is abilities or specials who are checked
+	 */
+	unit_ability_list overwrite_special_checking(const std::string& ability, unit_ability_list temp_list, unit_ability_list abil_list, const std::string& filter_self = "filter_self") const;
```

Comment:
Is there a reason to pass `abil_list` and `temp_list` by value instead of by reference and returning void?  `abil_list` could also be a const reference if I'm not mistaken.

Reply:
temp_list is the onmy value to must be returned. for abil_list could be a const refrence, i must test that before.

Reply:
i cannot compile with abil_list like const refrence

Reply:
abil_list is const refrence now, but i will not use void.

Reply:
You have `const unit_ability_list abil_list`, which should be `const unit_ability_list& abil_list` (missing `&`).